File: java/linkedlist/LinkedList.java
class Node {
    
        int data ;
        Node next ;
        Node(int data ){
            this.data = data ;
            this.next = null ;
        }
}
class InnerLinkedList {
    Node head ;
    InnerLinkedList(){
        this.head =null ;
    }
    
    void insert(int item ){
        Node newNode = new Node(item);
        if(head ==null){
            head = newNode ;
            return ;
        }
        Node current = head ; 
        while (current.next != null) {
            current =current.next ;
        }
        current.next = newNode ;

    }
    void display(){
        if (head ==null) {
            System.out.println("list is empty  ");
        }
        Node current = head ;

        while (current != null) {
            System.out.print(current.data+" ->");
            current= current.next ;
        }
    }

    void search(int data){
        Node current= head;
        
        while (current!= null) {
            if (current.data == data) {
                System.out.println("number is found ");
                return ;
            }
            current=current.next ;
        }
    }

    

 }


public class LinkedList {
    public static void main (String [] args ){
        InnerLinkedList list = new InnerLinkedList();
        list.insert(5);
        list.insert(10);
        list.insert(15);
        list.search(5);
        list.search(10);
        list.search(15);
        list.display();
    }
}

File: dotnet/LinkedList/LinkedList.cs
namespace LinkedLists;

public class LinkedList
{
    Node? head;
    public void insert(int num)
    {
        Node? newNode = new Node(num);
        if (head == null)
        {
            head = newNode;
            return;
        }

        Node Current = head;

        while (Current.next != null)
        {
            Current = Current.next;
        }

        Current.next = newNode;
    }

    public void Display()
    {
        Node? Current = head;
        while (Current != null)
        {
            Console.Write($"{Current.data} -> ");
            Current = Current.next;
        }
        Console.WriteLine();
    }

    public void update(int data, int newdata)
    {
        Node? Current = head;

        while (Current != null)
        {
            if (Current.data == data)
            {
                Current.data = newdata;
                return;
            }

            Current = Current.next;
        }
    }

     public void Delete(int data)
{
    if (head == null)
    {
        return;
    }

    if (head.data == data)
    {
        head = head.next;
        return;
    }

    Node current = head;

    while (current.next != null)
    {
        if (current.next.data == data)
        {
            current.next = current.next.next;
            return;
        }

        current = current.next;
    }

    Console.WriteLine("Value not found.");
}
}